GTG/core/timer.py

```diff
@@ -82,3 +86,35 @@ def set_configuration(self, refresh_hour, refresh_min):
 
     def get_configuration(self):
         return self.config.get('hour'), self.config.get('min')
+
+    def time_parser(self, time):
+        hour_value = ''
+        minute_value = ''
+        # For clarity, I'm using two match criteria:
+        # Format for HH:MM and optional am/pm
+        match_1 = re.match('([0-9]{1,2})(:)([0-9]{2})( )?([ap]m)?', time)
+        # Format for HH and am/pm
+        match_2 = re.match('([0-9]{1,2})( )?([ap]m)', time)
```

Comment:
Please, mark all regular expressions with as `r'<expression>'` (it modifies behaviour of backslashes)